### ISSUE TITLE
Incorrect value in Unicode_2_0, block=arabicpresentationformsb interval  "\ufe70\ufeff"

### ISSUE THREAD

**Issue by regisd** [jflex-de/jflex#835]
While generating Unicode_2_0.java with the ucd_generator, I observe a change in the output.

> 1) emitUnicodeVersionXY_2_0_14(de.jflex.ucd_generator.UcdGeneratorIntegrationTest)
org.junit.ComparisonFailure: intervals block=arabicpresentationformsb: expected:<\ufe70\ufef[e]> but was:<\ufe70\ufef[f]>

In https://www.unicode.org/Public/2.0-Update/Blocks-1.txt
```
FE50; FE6F; Small Form Variants
FE70; FEFF; Arabic Presentation Forms-B
FF00; FFEF; Halfwidth and Fullwidth Forms
```

Why should the interval be [L2202](https://github.com/jflex-de/jflex/blob/master/jflex/src/main/java/jflex/core/unicode/data/Unicode_2_0.java#L2202)
```
    // Unicode 2.0 property value: {block=arabicpresentationformsb}
    "\ufe70\ufefe",
```

**Comment by lsf37** [comment]
You're right, this looks wrong to me in the original. If I try tools like [this](https://util.unicode.org/UnicodeJsps/list-unicodeset.jsp?a=%5B%3Ablock%3Darabicpresentationformsb%3A%5D&g=&i=) they include `ufeff` (although that this one refers to a more recent unicode version, not 2.0).

@sarowe do you have any insights on this?

**Comment by regisd** [comment]
I'm now migrating the unicode tests, and I get a similar failure
> testBlocks(de.jflex.testcase.unicode.unicode_2_0.UnicodeBlocksTest_2_0)
java.lang.AssertionError: Not true that <[0000..007F; Basic Latin, 0080..00FF; **…** ; Halfwidth and Fullwidth Forms, FFF0..FFFD; Specials]> 
> contains all of <[0000..007F; Basic Latin, 0080..00FF; **…**; Small Form Variants, FE70..FEFF; Arabic Presentation Forms-B, FEFF..FEFF; Specials, FF00..FFEF; Halfwidth and Fullwidth Forms, FFF0..FFFF; Specials]>.
> It is missing <[FE70..FEFF; Arabic Presentation Forms-B, FFF0..FFFF; Specials]>

**Comment by sarowe** [comment]
TL;DR: This is a bug in the Unicode data file, rather than a bug in the JFlex implementation.

I'm not sure why, but the history for Unicode_2_0.java is truncated in master on github.  IntelliJ found the relevant commit for me: https://github.com/jflex-de/jflex/commit/3fdbd76 .  From the log for that commit:

> - Unicode 2.0's Blocks-X.txt has an error: character U+FEFF is assigned to two different blocks.  The assignment is removed from one of these blocks in UnicodeVersion.addInterval(), to reflect the fix that was made for this error in all following Unicode versions' Blocks(-X|-X.X.X).txt

See this error in https://www.unicode.org/Public/2.0-Update/Blocks-1.txt :

```
FE70; FEFF; Arabic Presentation Forms-B
[...]
FEFF; FEFF; Specials
```

See how it's fixed in the next Unicode version, from https://www.unicode.org/Public/2.1-Update4/Blocks-2.txt :

```
FE70; FEFE; Arabic Presentation Forms-B
FEFF; FEFF; Specials
```

**Comment by sarowe** [comment]
A little more context from the now-deleted jflex-unicode-maven-plugin: https://github.com/jflex-de/jflex/blob/cce73310960bafb1bbe9c04b57b41e8dcd5ca970/jflex-unicode-maven-plugin/src/main/java/jflex/maven/plugin/unicode/UnicodeVersion.java#L372-L381

**Comment by regisd** [comment]
Thanks for digging in. That explains this hack!
